lib/uplink/list_test.go

```diff
@@ -0,0 +1,233 @@
+	listChallenges []listChallenge // set of list commands and expected results
+}
+
+// TestPutGetList allows you to create a bucket, a set of paths, and a set of
+// list challenges that you want to try against that configuring. It will create
+// a unique file for each path, upload it, download it, and run the list
+// challenges against it.
+func TestPutGetList(t *testing.T) {
+	testplanet.Run(t, testplanet.Config{
+		SatelliteCount:   1,
+		StorageNodeCount: 1,
+		UplinkCount:      1},
+		func(t *testing.T, ctx *testcontext.Context, planet *testplanet.Planet) {
+			apiKey := planet.Uplinks[0].APIKey[planet.Satellites[0].ID()]
+			satelliteAddr := planet.Satellites[0].Local().Address.Address
+
+			tests := []putGetListTest{
+				{
+					bucket: "bu1",
+					paths: []string{
+						"a-file",
+						"a/b-file",
+						"a/b/slash-file",
+						"a/b////",
+						"a/b//",
+						"a/b//c-file",
+						"a/b//c/",
+						"//bob",
+						"/",
+					},
+					listChallenges: []listChallenge{
+						{
+							commands: []string{"/"},
+							expectedResults: []expectedResult{
+								{
+									path:     "/",
+									isPrefix: true,
+								},
+								{
+									path:     "",
+									isPrefix: false,
+								},
+							},
+						},
+						{
+							commands: []string{"//"},
+							expectedResults: []expectedResult{
+								{
+									path:     "bob",
+									isPrefix: false,
+								},
+							},
+						},
+						{
+							commands: []string{"a", "a/"},
+							expectedResults: []expectedResult{
+								{
+									path:     "b/",
+									isPrefix: true,
+								},
+								{
+									path:     "b-file",
+									isPrefix: false,
+								},
+							},
+						},
+						{
+							commands: []string{"a/b", "a/b/"},
+							expectedResults: []expectedResult{
+								{
+									path:     "/",
+									isPrefix: true,
+								},
+								{
+									path:     "slash-file",
+									isPrefix: false,
+								},
+							},
+						},
+						{
+							commands: []string{"a/b//"},
+							expectedResults: []expectedResult{
+								{
+									path:     "c/",
+									isPrefix: true,
+								},
+								{
+									path:     "c-file",
+									isPrefix: false,
+								},
+								{
+									path:     "/",
+									isPrefix: true,
+								},
+								{
+									path:     "",
+									isPrefix: false,
+								},
+							},
+						},
+						{
+							commands: []string{"a/b///"},
+							expectedResults: []expectedResult{
+								{
+									path:     "/",
+									isPrefix: true,
+								},
+							},
+						},
+						{
+							commands: []string{"a/b////"},
+							expectedResults: []expectedResult{
+								{
+									path:     "",
+									isPrefix: false,
+								},
+							},
+						},
+					},
+				},
+			}
+
+			for _, test := range tests {
+				runTest(ctx, t, apiKey, satelliteAddr, test)
+			}
+		})
+}
+
+func runTest(ctx context.Context, t *testing.T, apiKey, satelliteAddr string,
```

Comment:
I would rename this function to be `runTestPutGetList` to easy see that it's the companion of `TestPutGetList` and if we start to write more tests in this way in this package this function won't be the one with too generic name.